eve/main.yml

```diff
@@ -2671,3 +2686,42 @@ stages:
           name: Trigger multiple-nodes step with snapshot enabled
           stage_names:
             - multiple-nodes
+
+  publish-images:
+    worker: *build_worker
+    steps:
+      - ShellCommand: *wait_for_docker
+      - ShellCommand: *image_registry_login
+      - Git: *git_pull
+      - SetPropertyFromCommand:
+          name: Set shell-ui version as a property
+          property: shell_ui_version
+          command: |
+            . ./VERSION
+            echo "$VERSION_MAJOR.$VERSION_MINOR.$VERSION_PATCH$VERSION_SUFFIX"
+      - ShellCommand:
+          name: Build shell-ui container image
+          command: docker build . --tag shell-ui:v%(prop:shell_ui_version)s
+          workdir: build/shell-ui
+          haltOnFailure: true
+      - ShellCommand:
+          name: Publish shell-ui container image
+          command: |
+            if [ "$USE_PRODUCTION_REGISTRY" = true ]; then
+                PROJECT='%(secret:harbor_prod_project)s'
+                LATEST_STABLE_RELEASE=$(git tag | grep -E '^[0-9]+\.[0-9]+\.[0-9]+$' | sort -V | tail -n 1)
+                if [ "%(prop:shell_ui_version)s" = "$LATEST_STABLE_RELEASE" ]; then
+                    docker tag shell-ui:v%(prop:shell_ui_version)s shell-ui:stable
```

Comment:
When using USE_PRODUCTION_REGISTRY we might want to tag the existing image from dev harbor project instead of tagging a freshly built one.

Reply:
Since we're only pushing the development image in post-merge, it means during a release we would need to force a post-merge (or be sure nothing has been changed since the last post-merge ran) to publish a new production image.
It could work for this specific case, but I'm not sure that's a good thing to do as we're not really building the product from the  specific tag.
That said, I agree that's not perfect to rebuild a new image, which is not really tested (but it would also be the case if we take the one generated in post-merge).
It would probably be better to just publish the image used for integration tests, but it's not possible as it done now.